tests/app/questionnaire/routing/test_operators.py

```diff
@@ -314,6 +315,64 @@ def test_operation_date(date_string: str, offset):
     assert operator.evaluate(operands) == expected_result
 
 
+@pytest.mark.parametrize(
+    "offset, expected_result",
+    [
+        # Last Thursday (Yesterday from reference date)
+        ({"day_of_week": "THURSDAY"}, datetime(year=2020, month=12, day=31)),
+        # Last Saturday
+        ({"day_of_week": "SATURDAY"}, datetime(year=2020, month=12, day=26)),
+        # Last Week Thursday
+        (
+            {"day_of_week": "THURSDAY", "days": -7},
+            datetime(year=2020, month=12, day=24),
+        ),
+        # Last Week Saturday (Same as Last Saturday since reference date is a Friday)
+        (
+            {"day_of_week": "SATURDAY", "days": -7},
+            datetime(year=2020, month=12, day=26),
+        ),
+        # Next Thursday / Next Week Thursday
+        ({"day_of_week": "THURSDAY", "days": 7}, datetime(year=2021, month=1, day=7)),
+        # Next Saturday (Tomorrow from reference date)
+        ({"day_of_week": "SATURDAY", "days": 7}, datetime(year=2021, month=1, day=2)),
+        # Next Week Saturday
+        ({"day_of_week": "SATURDAY", "days": 14}, datetime(year=2021, month=1, day=9)),
+        # 1 full week Thursday
```

Comment:
1 full week from Thursday but an offset of 14 days, unfortunately I also find this confusing

Reply:
Ya, that is a consequence of us always get the "past" date first. We always resolve the last "day of the week", so we will be 1 week behind. No easy solution to this, whatever we do, this will exist in some form. If we didn't resolve to the past day always that brings its own complications also (initial did that).